contracts/AelinPool.sol

```diff
@@ -97,18 +114,61 @@ contract AelinPool is AelinERC20, MinimalProxyFactory, IAelinPool {
+
+        if (tmpAllowListAddresses.length > 0 || tmpAllowListAmounts.length > 0) {
             require(
-                _poolData.allowListAddresses.length == _poolData.allowListAmounts.length,
+                tmpAllowListAddresses.length == tmpAllowListAmounts.length,
                 "allowListAddresses and allowListAmounts arrays should have the same length"
             );
-            for (uint256 i = 0; i < _poolData.allowListAddresses.length; i++) {
-                allowList[_poolData.allowListAddresses[i]] = _poolData.allowListAmounts[i];
-                emit AllowlistAddress(_poolData.allowListAddresses[i], _poolData.allowListAmounts[i]);
+            for (uint256 i = 0; i < tmpAllowListAddresses.length; i++) {
+                allowList[tmpAllowListAddresses[i]] = tmpAllowListAmounts[i];
+                emit AllowlistAddress(tmpAllowListAddresses[i], tmpAllowListAmounts[i]);
             }
             hasAllowList = true;
         }
 
+        NftData[] memory tmpNftData = _poolData.nftData;
+
+        if (tmpNftData.length > 0) {
+            // if the first address supports punks or 721, the entire pool only supports 721 or punks
+            if (tmpNftData[0].collectionAddress == CRYPTO_PUNKS || NftCheck.supports721(tmpNftData[0].collectionAddress)) {
+                for (uint256 i = 0; i < tmpNftData.length; i++) {
+                    require(
+                        tmpNftData[i].collectionAddress == CRYPTO_PUNKS ||
+                            NftCheck.supports721(tmpNftData[i].collectionAddress),
+                        "can only contain 721"
+                    );
+                    nftCollectionDetails[tmpNftData[i].collectionAddress] = tmpNftData[i];
+                    emit PoolWith721(
+                        tmpNftData[i].collectionAddress,
+                        tmpNftData[i].purchaseAmount,
+                        tmpNftData[i].purchaseAmountPerToken
+                    );
+                }
+            }
+            // if the first address supports 1155, the entire pool only supports 1155
+            if (NftCheck.supports1155(tmpNftData[0].collectionAddress)) {
+                for (uint256 i = 0; i < tmpNftData.length; i++) {
+                    require(NftCheck.supports1155(tmpNftData[i].collectionAddress), "can only contain 721");
+                    nftCollectionDetails[tmpNftData[i].collectionAddress] = tmpNftData[i];
+
+                    for (uint256 j = 0; j < tmpNftData[i].tokenIds.length; j++) {
+                        nftId[tmpNftData[i].collectionAddress][tmpNftData[i].tokenIds[j]] = true;
+                    }
+                    emit PoolWith1155(
+                        tmpNftData[i].collectionAddress,
+                        tmpNftData[i].purchaseAmount,
+                        tmpNftData[i].purchaseAmountPerToken,
+                        tmpNftData[i].tokenIds,
+                        tmpNftData[i].minTokensEligible
```

Comment:
this is not exactly right. we need the minTokensEligible per ID

Reply:
E.g. ID 0 is a sword where there are 100,000 total and ID 1 is a shield with 1,000 total. You can say that anyone with 1000 swords can participate or anyone with 10 shields. The numbers can be different per ID. We never blacklist the ERC1155 IDs though. that is one thing to keep in mind as well

Reply:
Perhaps we can blacklist an address but I think we just want to let 1155 collection holders mint unlimited amounts and not try to control them. It is too easily gamed

Reply:
yea sure. will change `minTokensEligible` into an array

Reply:
added. 5e0426fb0e68557fda5f49de0412911a8bbed85b